### ISSUE TITLE
LFX workspace: OpenCV SDKs for Wasm

### ISSUE THREAD

**Issue by juntao** [WasmEdge/WasmEdge#1747]
## Motivation

WasmEdge is a leading WebAssembly runtime for AI inference. It supports AI frameworks such as Tensorflow, OpenVINO and PyTorch. A compelling use case is computer vision applications on the edge.

Computer vision applications need to pre-process images and videos into tensor formats before applying the AI model. They then often need to overlay the tensor results onto the original image. In our existing demos, we use the Rust [image crate](https://crates.io/crates/image) to process images. However, the crate only has limited features and is inadequate for many computer vision applications.

In the Python-based computer vision applications, the image pre-processing is often done with the Python wrapper for OpenCV library. The OpenCV library itself is written in C and can be compiled into WebAssembly. We would like to create an OpenCV SDK that allows WebAssembly applications to call OpenCV functions.

## Details

* Study how OpenCV is [compiled into WebAssembly through the EMCC](https://www.npmjs.com/package/opencv-wasm), and its JavaScript API for web apps.
* Attempt to compile OpenCV and its [Rust wrapper](https://github.com/twistedfall/opencv-rust) crate into WebAssembly (refer to the Rust wrapper to [compile QuickJS](https://github.com/second-state/wasmedge-quickjs) into WebAssembly)
* Create computer vision examples using the OpenCV library in WasmEdge
* Benchmark the AOT version OpenCV in WasmEdge against the Python version
* Optional: Create a JavaScript wrapper for OpenCV in WasmEdge QuickJS

## Appendix

* The [opencv-wasm](https://www.npmjs.com/package/opencv-wasm) package for node.js
* [Rust binding for OpenCV](https://github.com/twistedfall/opencv-rust)
* [WasmEdge tensorflow SDK](https://wasmedge.org/book/en/dev/rust/tensorflow.html)
* [WASI-NN examples for WasmEdge](https://github.com/second-state/WasmEdge-WASINN-examples)

**Comment by alabulei1** [comment]
> Hi @juntao can i work on this ?

Hey @omkar-mohanty, We'd like to ask you to complete a pretest for better collaboration. See details #1754. Thank you.

**Comment by dannypsnl** [comment]
Summary OpenCVMini

## Feature
### transform
1. `normalize`: not `cv::normalize`, from https://github.com/WasmEdge/WasmEdge/commit/77051da4995d7318d91a82102a72ce2557151764#diff-3333d926ca87cf4285bfcd6deae45ee310307be66fca8a4ca6f0f8a946743fccR50-R54
2. `blur`: blurring function
3. `bilinear_sampling`: bilinear sampler
### read/write
1. `imdecode`: from image from instance buffer
2. `imencode`: put image into instance buffer
4. `imwrite`: write image into file
### misc
1. `imshow`: show image (broken on MacOS)
2. `waitkey`: wait any keypress

## Example

- Repository: https://github.com/second-state/opencvmini-example

**Comment by dannypsnl** [comment]
Due to PR merged, this can be seem as complete.
